libqtile/core/manager.py

```diff
@@ -530,19 +539,53 @@ def current_window(self):
     def add_strut(self, strut):
         from libqtile.bar import Bar, Gap
 
+        try:
+            if strut[0]:
+                x = strut[0]
+            elif strut[1]:
+                x = self.x_screen_dimensions.get('x') - strut[1]
+            else:
+                x = ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+            if strut[2]:
+                y = strut[2]
+            elif strut[3]:
+                y = self.x_screen_dimensions.get('y') - strut[2]
+            else:
+                y = ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            screen = self.find_screen(x, y)
+            assert screen is not None
+            assert screen.x is not None and screen.y is not None and \
+                   screen.height is not None and screen.width is not None
+        except AssertionError:  # strut Values are defined but the coordinates are not to be found on any screen
```

Comment:
I still think we can drop all the asserts in favor of if. It'll let you drop a level of indentation here.